src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/ItemManager.cs

```diff
@@ -570,6 +583,21 @@ private static RoslynCompletionItem GetOrAddRoslynCompletionItem(VSCompletionIte
             return roslynItem;
         }
 
+        // If the item didn't match the filter text, we still keep it in the list
+        // if one of two things is true:
+        //  1. The user has typed nothing or only typed a single character.  In this case they might
+        //     have just typed the character to get completion.  Filtering out items
+        //     here is not desirable.
+        //
+        //  2. They brought up completion with ctrl-j or through deletion.  In these
+        //     cases we just always keep all the items in the list.
+        private static bool KeepAllItemsInTheList(CompletionTriggerKind initialTriggerKind, string filterText)
+        {
+            return filterText.Length <= 1 ||
+                initialTriggerKind == CompletionTriggerKind.Invoke ||
+                initialTriggerKind == CompletionTriggerKind.Deletion;
```

Comment:
i wish this was shared with some of our other logic elsewhere so that this would stay consistent with that if we ever change things.

Reply:
You mean LSP?